bpfman/src/bpf.rs

```diff
@@ -54,22 +50,31 @@ const MAPS_USED_BY_PREFIX: &str = "map_used_by_";
 pub(crate) struct BpfManager {
     config: Config,
     commands: Option<Receiver<Command>>,
-    pub(crate) image_manager: Option<Sender<ImageManagerCommand>>,
+    pub(crate) image_manager: Option<ImageManager>,
 }
 
 impl BpfManager {
-    pub(crate) fn new(
-        config: Config,
-        commands: Option<Receiver<Command>>,
-        image_manager: Option<Sender<ImageManagerCommand>>,
-    ) -> Self {
+    pub(crate) fn new(config: Config, commands: Option<Receiver<Command>>) -> Self {
         Self {
             config,
             commands,
-            image_manager,
+            image_manager: None,
         }
     }
 
+    // Make sure to call init_image_manger if the command requires interation with
+    // an OCI based container registry. It should ONLY be used where needed, to
+    // explicitly control when bpfman blocks for network calls to both sigstore's
+    // cosign tuf registries and container registries.
+    pub(crate) async fn init_image_manager(&mut self) {
+        let config = open_config_file();
+        self.image_manager = Some(
```

Comment:
This should only happen if `self.image_manager.is_none()`.
I know you've put comments here, but humans make mistakes.

Reply:
Fair enough, done!